main.py

```diff
@@ -365,46 +399,107 @@ def performEvaluation(issue_data) -> Result:
+            result.set_preservation_status("Build script missing") 
+            return result
+        
+        gradle_files_destination_path = os.path.join(issue_folder_abs_dir, issue_id, specimin_output, repo_name)
 
-    if not os.path.exists(build_gradle_path) or not os.path.exists(settings_gradle_path):
-        print(f"{issue_id}: {build_gradle_path} or {settings_gradle_path} not found.")
-        result.set_preservation_status("Build script missing") 
-        return result
-    
-    gradle_files_destination_path = os.path.join(issue_folder_abs_dir, issue_id, specimin_output, repo_name)
+        copy_command = f"cp {build_gradle_path} {settings_gradle_path} {gradle_files_destination_path}"
+        subprocess.run(copy_command, shell=True)
+        
+        #../ISSUES/cf-xx/output/projectname/build_log.txt
+        log_file = os.path.join(issue_folder_abs_dir, issue_id, specimin_output, repo_name, minimized_program_build_log_file)
+
+        if os.path.exists(log_file):
+            os.remove(log_file)
+
+        target_gradle_script = os.path.join(gradle_files_destination_path, "build.gradle")
+        # Open the log file in write mode
+        min_prgrm_build_status = None
+        with open(log_file, "w") as log_file_obj:
+            min_prgrm_build_status = subprocess.run(f"./gradlew -b  {target_gradle_script} compileJava", cwd = specimin_path, shell=True, stderr=log_file_obj)
+            print(f"{issue_id} Minimized program gradle build status = {min_prgrm_build_status.returncode}")
+        if min_prgrm_build_status.returncode == 0:
+            print(f"{issue_id} Minimized program gradle build successful. Expected: Fail")
+            result.set_preservation_status("Target behavior is not preserved.")
+            return result
+    else: 
+        #TODO: some targets don't reproduce target property with gradle build. 
+        #Build them with shell
+        existing_jdk_dir = os.environ.get("JAVA_HOME")
+        cf_url = issue_data.get("cf_release_url", "")
+        version = issue_data.get("cf_version", "1.9.13")
+        cf_path = f"checker-framework-{version}"
+        cf_abs_path = os.path.abspath(cf_path)
+        cf_zip = f"{cf_abs_path}.zip"
+        full_url = cf_url + "/" + cf_path + "/" + cf_path + ".zip"
+        download_with_wget(full_url, cf_zip)
+        if os.path.exists(cf_zip):
+            unzip_file(cf_zip)
+        
+        jdk_template_url = "https://corretto.aws/downloads/latest/amazon-corretto-{version}-{arch}-{os}-jdk.tar.gz"
+        version = issue_data.get(JsonKeys.JAVA_VERSION.value, "11")
+        #TODO: if version is not 8, no need to pull jdk
+        arch = "x64"
+        if platform.machine() == "arm64": #ignoring x86
+            arch = "aarch64"
+        op = "linux"
+        if platform.system() == "Darwin":
+            op = "macos"
+        elif platform.system() == "Windows":
+            print("windows not supported") #TODO: return or exception
```

Comment:
This should definitely raise an exception rather than silently continuing.

Reply:
I added this check in main method so that if fails at the start of the program.